extensions/unsplash/package.json

```diff
@@ -0,0 +1,170 @@
+      "mode": "no-view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "accessKey",
+      "title": "Access Key",
+      "description": "Get your API key from https://unsplash.com/app/api",
+      "type": "password",
+      "required": true
+    },
+    {
+      "name": "username",
+      "title": "Username",
+      "placeholder": "Your Unsplash username",
+      "description": "Your username on Unsplash. Needed to list your lists.",
+      "type": "textfield",
+      "required": false
+    },
+    {
+      "name": "orientation",
+      "title": "Image Orientation",
+      "description": "The orientation of images that will be displayed as result.",
+      "type": "dropdown",
+      "data": [
+        {
+          "title": "All",
+          "value": "all"
+        },
+        {
+          "title": "Landscape",
+          "value": "landscape"
+        },
+        {
+          "title": "Portrait",
+          "value": "portrait"
+        },
+        {
+          "title": "Squarish",
+          "value": "squarish"
+        }
+      ],
+      "required": false,
+      "default": "all"
+    },
+    {
+      "name": "downloadSize",
+      "title": "Default Download Size",
+      "description": "The default size of the image to download. This might not be effective on every image, it will be applied where possible.",
+      "type": "dropdown",
+      "data": [
+        {
+          "title": "Original",
+          "value": "raw"
+        },
+        {
+          "title": "Full",
+          "value": "full"
+        },
+        {
+          "title": "Regular",
+          "value": "regular"
+        },
+        {
+          "title": "Small",
+          "value": "small"
+        }
+      ],
+      "required": false,
+      "default": "full"
+    },
+    {
+      "name": "collections",
+      "title": "Custom Collections",
+      "placeholder": "Collection IDs, comma & space separated",
+      "description": "Leave blank for pre-selected ones - Collections to search for images on Set Random Wallpaper. Comma & space separated list of collection IDs.",
+      "type": "textfield",
+      "required": false,
+      "default": ""
+    },
+    {
+      "name": "includeDefaults",
+      "title": "Include Default Collections",
+      "type": "dropdown",
+      "description": "If you're using custom collections, select whether to include the default ones or not.",
+      "data": [
+        {
+          "title": "Yes",
+          "value": "yes"
+        },
+        {
+          "title": "No",
+          "value": "no"
+        }
+      ],
+      "required": false,
+      "default": "no"
+    },
+    {
+      "name": "path",
+      "title": "Save Path",
+      "description": "Location that the downloaded images will be set",
+      "type": "textfield",
+      "placeholder": "/Pictures/Wallpapers (defaults to ApplicationSupport)",
+      "required": false,
+      "default": ""
+    }
```

Comment:
This setting confused me a lot, maybe this should be a fullpath instead (With tilde support) and leave `(defaults to ApplicationSupport)` since the text is too long for the field.

When I use `Download Image` and have set a custom path, I would expect it to download it to this folder

Reply:
Since almost any file that are not in the user directory is not accessible without sudo permissions, I wanted to prepend home directory to this so it gets shorter and you don't have to deal with permissions. Tried tilde support with `fs.resolve` in the beginning, didn't work for some reason, am I missing something?

Reply:
Is there a need of this setting then? If the `Download Image` command always pop `Select output folder` window.

By the way, I just used [expand-tilde](https://www.npmjs.com/package/expand-tilde) last time I had to 😄

Reply:
You're right, but it is more of "Default Download Location" in case users want to save all of the downloaded images outside of Application Support, maybe a Wallpaper folder on their Documents even. When they run Set Desktop Wallpaper it basically downloads the image to that given directory. Do you think it is bad, or unnecessary?

Reply:
I would expect if I set the `Save Path` that `Download Image` would save to this folder, if blank, pop `Select output folder` – I would not expect it to save it to a folder specific folder when it sets the wallpaper

Reply:
But the thing is, as myself needed it this way; there are times I want to download the image to a different location. Like, most of the time, especially when I'm changing wallpapers, I don't want to change the path all the time, nor I want to fill up my Desktop or Wallpapers directory with those images. So I made the "Download Image" option to *download to a selected location* whenever I want that image on somewhere else on my PC (instead of Application Support for instance). Do you think this is bad and should I remove it?

Reply:
Why not just always download it to `~/Downloads`?

Reply:
@eggsy what is this setting used for? Wouldn't it be the easiest to just always download the images to the `~/Downloads` folder? At least the once that you want to keep. If you want to go the extra mile, you could have another action called "Download to Location" which then let's you select where you want to store the image. This could pop a form or something like that. WDYT?

For the command that sets the wallpaper, I'd suggest to just download it to the extension support directory. The directory is meant for this kind of things. I don't think you need to have a setting for this.

Reply:
I will then delete this setting and always download images to Application Support, and rename Download Image to Download to Location, does that sound good?

Reply:
`Download Image` sounded better IMO – but you decide 😄